dspace-api/src/test/java/org/dspace/discovery/DiscoveryTest.java

```diff
@@ -0,0 +1,124 @@
+
+
+    WorkspaceItem leftIs;
+    WorkspaceItem rightIs;
+
+    /**
+     * This method will be run before every test as per @Before. It will
+     * initialize resources required for the tests.
+     *
+     * Other methods can be annotated with @Before here or in subclasses
+     * but no execution order is guaranteed
+     */
+    @Before
+    @Override
+    public void init() {
+        super.init();
+        try {
+            context.turnOffAuthorisationSystem();
+            Community community = communityService.create(null, context);
+
+            Collection col = collectionService.create(context, community);
+            leftIs = workspaceItemService.create(context, col, false);
+            rightIs = workspaceItemService.create(context, col, false);
+
+            itemService.addMetadata(context, leftIs.getItem(), "dc", "description", "abstract", null, "headache");
+            itemService.addMetadata(context, rightIs.getItem(), "dc", "description", "abstract", null, "headache");
+
+            workspaceItemService.update(context, leftIs);
+            workspaceItemService.update(context, rightIs);
+
+            indexingService.commit();
+
+            context.restoreAuthSystemState();
+            context.dispatchEvents();
+        } catch (AuthorizeException ex) {
+            log.error("Authorization Error in init", ex);
+            fail("Authorization Error in init: " + ex.getMessage());
+        } catch (SQLException ex) {
+            log.error("SQL Error in init", ex);
+            fail("SQL Error in init: " + ex.getMessage());
+        } catch (Exception ex) {
+            log.error("Error in init", ex);
+            fail("Error in init: " + ex.getMessage());
+        }
+    }
+
+    /**
+     * This method will be run after every test as per @After. It will
+     * clean resources initialized by the @Before methods.
+     *
+     * Other methods can be annotated with @After here or in subclasses
+     * but no execution order is guaranteed
+     */
+    @After
+    @Override
+    public void destroy() {
+        context.abort();
+        super.destroy();
+    }
+
+    @Test
+    public void deleteWorkspaceItemSolrRecordAfterDeletionFromDbTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        DiscoverQuery discoverQuery = new DiscoverQuery();
+        discoverQuery.setQuery("*:*");
+        discoverQuery.addFilterQueries("search.resourceid:" + leftIs.getID());
+        DiscoverResult discoverResult = searchService.search(context, discoverQuery);
+        List<IndexableObject> indexableObjects = discoverResult.getIndexableObjects();
+        assertTrue(indexableObjects.size() == 1);
+        workspaceItemService.deleteAll(context, leftIs);
+        discoverResult = searchService.search(context, discoverQuery);
+        indexableObjects = discoverResult.getIndexableObjects();
+        assertTrue(indexableObjects.size() == 0);
```

Comment:
I'd recommend also changing this to an `assertEquals` statement...as if it is ever *not* zero, we'd want information around what it's value actually is.